kratos/solving_strategies/schemes/scheme.h

```diff
@@ -756,6 +791,17 @@ class Scheme
      * @param EquationId The ID's of the condition degrees of freedom
      * @param rCurrentProcessInfo The current process info instance
      */
+    virtual void CalculateSystemContributions(
```

Comment:
Those functions itself can also be `const` to my understanding, right?
I mean, they should not modify contents of the `Scheme` (if they do, it would be done from inside a parallel section and could be a race-condition

Reply:
Agree. However this would require to also add the `const` to all the `override`'s in the derived classes (we can leave it for a future PR)

Reply:
I did it right away, was easier